src/observer.ts

```diff
@@ -362,21 +398,26 @@ export class DataLayerObserver {
 
     try {
       const target = DataLayerTarget.find(source);
-      this.registerTarget(target, operators, destination, readOnLoad, monitor, debug, debounce);
-    } catch (_) {
-      // schedule subsequent attempts at (attempt * wait) later
-      setTimeout(() => {
-        try {
-          const target = DataLayerTarget.find(source);
-          this.registerTarget(target, operators, destination, readOnLoad, monitor, debug, debounce);
-        } catch (err) {
-          if (attempt > 3) {
-            Logger.getInstance().error(LogMessageType.RuleRegistrationError, { rule: id, source, reason: err.message });
+
+      switch (typeof waitUntil) {
+        case 'number':
+          // NOTE this delay is scheduled *after* the data layer is found to be defined on the page (not after page load)
+          setTimeout(() => {
```

Comment:
Rather than re-use `registerWithDelay` if the delay is a specific number of milliseconds.  Trying to re-use `registerWithDelay` for this use case creates another call to `registerRule`, which is circuitous and likely unexpected.

Reply:
Check out this PR: https://github.com/fullstorydev/fullstory-data-layer-observer/pull/146

I've removed that circular dependancy.